packages/polling-controller/src/PollingController.ts

```diff
@@ -49,19 +59,26 @@ function PollingControllerMixin<TBase extends Constructor>(Base: TBase) {
      * Starts polling for a networkClientId
      *
      * @param networkClientId - The networkClientId to start polling for
+     * @param options - The options used to group the polling events
      * @returns void
      */
-    startPollingByNetworkClientId(networkClientId: NetworkClientId) {
+    startPollingByNetworkClientId(
+      networkClientId: NetworkClientId,
+      options: object = {},
+    ) {
       const innerPollToken = random();
-      if (this.#networkClientIdTokensMap.has(networkClientId)) {
-        const set = this.#networkClientIdTokensMap.get(networkClientId);
-        set?.add(innerPollToken);
+
+      const id = getKey(networkClientId, options);
+
+      const innerPollTokenSet = this.#networkClientIdTokensMap.get(id);
+      if (innerPollTokenSet) {
+        innerPollTokenSet.add(innerPollToken);
```

Comment:
[nit] since there is no `outerPollToken` now:
```suggestion
      const pollTokenSetForKey = this.#networkClientIdTokensMap.get(id);
      if (pollTokenSetForKey) {
        pollTokenSetForKey.add(pollToken);
```

Reply:
+1 for removing overhead. It would also be easier to understand if we make the name of this variable the same in both branches. So, something like this:

``` suggestion
      const set = this.#networkClientIdTokensMap.get(id);
      if (set) {
        set.add(pollToken);
```